File: frontend/tips/6-open-new-window/code/python/new_window_and_tab.py
import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait

@pytest.fixture
def driver_setup_teardown():
    driver = webdriver.Firefox()
    yield driver
    driver.quit()

def test_new_window(driver_setup_teardown):
    driver = driver_setup_teardown
    driver.get('https://the-internet.herokuapp.com/windows')
    driver.switch_to.new_window('window')
    driver.get('https://the-internet.herokuapp.com/typos')
    assert len(driver.window_handles) == 2

def test_new_tab(driver_setup_teardown):
    driver = driver_setup_teardown
    driver.get('https://the-internet.herokuapp.com/windows')
    driver.switch_to.new_window('tab')
    driver.get('https://the-internet.herokuapp.com/typos')
    assert len(driver.window_handles) == 2

File: frontend/tips/6-open-new-window/code/ruby/new_window_and_tab.rb
require 'selenium-webdriver'
require 'rspec/expectations'
include RSpec::Matchers

def setup
  @driver = Selenium::WebDriver.for(:firefox)
end

def teardown
  @driver.quit
end

def run
  setup
  yield
  teardown
end

run do
    @driver.get 'https://the-internet.herokuapp.com/windows'
    @driver.switch_to.new_window(:window)
    @driver.get('https://the-internet.herokuapp.com/typos')
    expect(@driver.window_handles.size).to eql 2
end

run do
    @driver.get 'https://the-internet.herokuapp.com/windows'
    @driver.switch_to.new_window(:tab)
    @driver.get('https://the-internet.herokuapp.com/typos')
    expect(@driver.window_handles.size).to eql 2
end
